test/src/reporters/ElementIDTest.C

```diff
@@ -0,0 +1,31 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ElementIDTest.h"
+
+registerMooseObject("MooseTestApp", ElementIDTest);
+
+InputParameters
+ElementIDTest::validParams()
+{
+  InputParameters params = ElementReporter::validParams();
+  params.addRequiredParam<ExtraElementIDName>("id_name1", "Name of an extra element id");
+  params.addRequiredParam<ExtraElementIDName>("id_name2", "Name of an extra element id");
+  return params;
+}
+
+ElementIDTest::ElementIDTest(const InputParameters & params)
+  : ElementReporter(params),
+    _id_name1(getParam<ExtraElementIDName>("id_name1")),
+    _id_name2(getParam<ExtraElementIDName>("id_name2")),
+    _mapping(declareValueByName<std::unordered_map<dof_id_type, std::set<dof_id_type>>>(
```

Comment:
This mode is going to fail sweeps. It only runs on rank 0, where your mapping is going to change on rank 0 with more procs.

Reply:
The map is constructed on all processors identically. Will this be ok?

Reply:
No. Elem IDs change with partitioning in the sweeps. You'll need to disable renumbering. 

Also - is it not concerning that this map could be huge? Not having a distributed version of this makes me uncomfortable.

Reply:
I was mostly wrong here, as I thought that the key was element ID. So we're good, this shouldn't fail.

However, the use of `REPORTER_MODE_ROOT` is incorrect here. It should be `REPORTER_MODE_REPLICATED`, as the method will replicate the data on all ranks for you.

Reply:
Do you want me push a quick update to replace `REPORTER_MODE_ROOT` with `REPORTER_MODE_REPLICATED`. I guess it is ok to leave it as it is because this is only a test object and no other object is requesting the map? Talked with Logan offline, seems we can leave this as it is. If we add more contents in this test object, we can update this together.